src/lib/es5.d.ts

```diff
@@ -210,6 +210,8 @@ interface ObjectConstructor {
       * Returns the names of the enumerable properties and methods of an object.
       * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
       */
+    keys<T>(o: Array<T>): string[];
+    keys<T extends { [key: string]: any }>(o: T): (keyof T & string)[];

```

Comment:
> adding the & string removes any type safety the literal types give you. 

I don't really understand that. Why would you lose type safety? The `& string` version still only allows certain strings, rather than all strings, which is more safe.

Let me give the real code example that motivated these two PRs:

```ts
const loggers = {
  info: ...,
  warn: ...,
  error: ...
}

Object.entries(loggers).forEach(([name, logger]) => {
  // In here, I want typeof name to be "info" | "warn" | "error", so that 
  // the line below will fail if I add a key to loggers by mistake that
  // doesn't have a corresponding method on console. (This in fact
  // happened: I originally had a debug logger, but only later realized
  // that there is no console.debug() on the node console object.)

  logger.log = console[name].bind(console);
})
```

To make that code work, `string[]` isn't enough. The change merged in #12207 meanwhile, works perfectly, but fails in the (somewhat obscure) case that motivated the `& string` addition in this PR. Using `& string`, the type of `name` is inferred as `("info" & string) | ("warn" & string) | ("error" & string)`. That should, if I understand correctly, be equivalent to `"info" | "warn" | "error"`. However, I instead get an error. That seems like a separate bug, though (opened issue #12289) for it, also tied to simplification. If #12289 were addressed, I think the `& string` would be the right type signature for `Object.keys/entries`.